app/models.py

```diff
@@ -43,21 +43,31 @@ def list():
         return [c.value for c in Taxonomies]
 
 
-def country_to_ISO_code(value: str):
+def alpha2_to_country_name(value: Optional[str]):
     """
-    Helper function that return ISO code for country
+    Helper function to return country name for aplha2 code
     """
-    country_data = pycountry.countries.get(name=value)
-    if country_data is not None:
-        country_iso_code = country_data.alpha_2
-        return f"{country_iso_code.lower()}-en"
+    if value is not None and len(value) == 2:
+        country = pycountry.countries.get(alpha_2=value)
+        if country is not None:
+            return f"{country.name}"
+    return value
+
+
+def country_name_to_alpha2(value: Optional[str]):
+    """
+    Helper function that return alpha2 code for country name
+    """
+    country = pycountry.countries.get(name=value)
+    if country is not None:
+        return f"{(country.alpha_2).lower()}-en"
```

Comment:
Why would you return - en ? It should depends on the language code (aka lc)

Imagine you are on fr.openfoodfacts.org, people would like to continue on frenche pages not on the english one.

So IMO, the API should support an optional lc parameter.

This function should use this parameter.
If country code == language code, just use "{country_code}" , else user "{country_code}-en" (because we don't support any language on any country) and if it's world use "world-{language_code}".

You can do this in a separate PR if you prefer.

Reply:
I will look into it.